libnopegl/src/node_texture.c

```diff
@@ -486,6 +490,67 @@ static int texture_update(struct ngl_node *node, double t)
     return 0;
 }
 
+static int rtt_resize(struct ngl_node *node)
+{
+    int ret = 0;
+    struct ngl_ctx *ctx = node->ctx;
+    struct texture_priv *s = node->priv_data;
+
+    const int32_t width = ctx->current_rendertarget->width;
+    const int32_t height = ctx->current_rendertarget->height;
+    if (s->params.width == width && s->params.height == height)
+        return 0;
+
+    struct texture *texture = ngli_texture_create(ctx->gpu_ctx);
+    if (!texture) {
+        ret = NGL_ERROR_MEMORY;
+        goto fail;
+    }
+
+    struct texture_params texture_params = s->params;
+    texture_params.width = width;
+    texture_params.height = height;
+
+    ret = ngli_texture_init(texture, &texture_params);
+    if (ret < 0)
+        goto fail;
+
+    struct rtt_ctx *rtt_ctx = ngli_rtt_create(ctx);
+    if (!rtt_ctx) {
+        ret = NGL_ERROR_MEMORY;
+        goto fail;
+    }
+
+    struct rtt_params rtt_params = s->rtt_params;
+    rtt_params.width  = width;
+    rtt_params.height = height;
+    rtt_params.colors[0].attachment = texture;
+
+    ret = ngli_rtt_init(rtt_ctx, &rtt_params);
+    if (ret < 0)
+        goto fail;
+
+    ngli_rtt_freep(&s->rtt_ctx);
+    ngli_texture_freep(&s->texture);
+
+    s->params = texture_params;
+    s->texture = texture;
+    s->image.params.width = width;
+    s->image.params.height = height;
+    s->image.planes[0] = texture;
+    s->image.rev = s->image_rev++;
+    s->rtt_params = rtt_params;
+    s->rtt_ctx = rtt_ctx;
+
+    return 0;
+fail:
```

Comment:
nit: can you add a line break before the label?

Reply:
Done.